docs/examples/ex_match.rst

```diff
@@ -0,0 +1,70 @@
+Pair Modelled and Observed Data
+===============================
+
+One common analytical need is to obtain a combined data set of modelled and
+observed observations at the same times and locations.  This can be done using
+the ``collect_inst_model_pairs`` routine in
+`utils.match <../utils.html#module-pysatModels.utils.match>`_.  This routine
+takes a date range as input, and then extracts modelled observations at the
+specified instrument location.  However, it does not interpolate in time.
+Details about the interpolation of modelled data onto the instrument location
+can be found in the
+`utils.extract <../utils.html#module-pysatModels.utils.extract>`_ routine:
+``extract_modelled_observations``.
+
+In the example below, we load a series of TIEGCM files that were created every
+15 minutes and are stored in a specified directory.  We pair them with CINDI
+C/NOFS ion drift data that was previously downloaded using pysat.
+
+::
+
+
+   import datetime as dt
+   import pysat
+   import pysatModels as ps_mod
+
+   stime = dt.datetime(2010, 1, 1)
+   etime = dt.datetime(2010, 2, 1)
+   tinc = dt.timedelta(minutes=15)
+   input_kwargs = dict()
+
+   # Initialize the model input information, including the keyword arguements
+   # needed to load the model Instrument into an xarray Dataset using
+   # the match.load_model_xarray routine
+   filename = 'sample_experiment/tiegcm_%Y%j.%H%M.nc'
+   tiegcm = pysat.Instrument(platform='ucar', name='tiegcm')
+   input_kwargs["model_load_kwargs"] = {'model_inst': tiegcm,'filename': filename}
+
+   # Initialize the CINDI C/NOFS input, and ensure the best model interpolation
+   # by extracting the clean model data after matching
+   cindi = pysat.Instrument(platform='cnofs', name='ivm', clean_level='none')
+   cindi.download(stime, etime) # Skip this if you already have the data
+   input_kwargs["inst_clean_rout"] = pysat.instruments.cnofs_ivm.clean
+   input_kwargs["inst_download_kwargs"] = {"skip_download": True}
+
+   # Many keyword arguements are required, as they provide information on
+   # the names of coordinates needed to extract data
+   input_kwargs["inst_lon_name"] = "glon"
+   input_kwargs["mod_lon_name"] = "longitude"
+   input_kwargs["inst_name"] = ["glon", "glat", "altitude"]
+   input_kwargs["mod_name"] = ["longitude", "latitude", "altitude"]
+   input_kwargs["mod_units"] = ["deg", "deg", "km"]
+   input_kwargs["mod_datetime_name"] = "time"
+   input_kwargs["mod_time_name"] = "time"
+
+   # Now we are ready to run using the defaults!
+   matched_inst = ps_mod.utils.match.collect_inst_model_pairs(stime, etime, tinc, cindi, **input_kwargs)
+
```

Comment:
Not working for me quite yet.

cindi[['glon', 'glat', 'altitude']]
Out[31]: 
                               glon      glat    altitude
2010-05-29 00:00:03.275  322.633667  7.099998  494.529999
2010-05-29 00:00:03.775  322.663269  7.106115  494.432648
2010-05-29 00:00:04.275  322.692871  7.112232  494.335297
2010-05-29 00:00:04.775  322.722473  7.118345  494.237976

In [34]: tiegcm[['lon', 'lat', 'ZG']]
Out[34]: 
<xarray.Dataset>
Dimensions:  (ilev: 57, lat: 72, lon: 144, time: 24)
Coordinates:
  * lon      (lon) float64 -180.0 -177.5 -175.0 -172.5 ... 172.5 175.0 177.5
  * lat      (lat) float64 -88.75 -86.25 -83.75 -81.25 ... 83.75 86.25 88.75
    LBC      float64 ...
  * ilev     (ilev) float64 -7.0 -6.75 -6.5 -6.25 -6.0 ... 6.0 6.25 6.5 6.75 7.0
  * time     (time) datetime64[ns] 2010-05-29T01:00:00 ... 2010-05-30
Data variables:
    ZG       (time, ilev, lat, lon) float32 ...


In [36]: matched_inst = ps_mod.utils.match.collect_inst_model_pairs(stime, etime, tinc, cindi, **input_kwargs)

In [37]: matched_inst

In [38]: 

Altitude in TIEGCM isn't like the others.

Reply:
Code setup:

In [38]:    import datetime as dt
    ...:    import pysat
    ...:    import pysatModels as ps_mod
    ...: 
    ...:    stime = dt.datetime(2010, 5, 29)
    ...:    etime = dt.datetime(2010, 5, 29)
    ...:    tinc = dt.timedelta(minutes=15)
    ...:    input_kwargs = dict()
    ...: 
    ...:    # Initialize the model input information, including the keyword arguements
    ...:    # needed to load the model Instrument into an xarray Dataset using
    ...:    # the match.load_model_xarray routine
    ...:    filename = 'sample_experiment/tiegcm_%Y%j.%H%M.nc'
    ...:    tiegcm = pysat.Instrument(platform='ucar', name='tiegcm')
    ...:    input_kwargs["model_load_kwargs"] = {'model_inst': tiegcm,'filename': filename}
    ...: 
    ...:    # Initialize the CINDI C/NOFS input, and ensure the best model interpolation
    ...:    # by extracting the clean model data after matching
    ...:    cindi = pysat.Instrument(platform='cnofs', name='ivm', clean_level='none')
    ...:    #cindi.download(stime, etime) # Skip this if you already have the data
    ...:    input_kwargs["inst_clean_rout"] = pysat.instruments.cnofs_ivm.clean
    ...:    input_kwargs["inst_download_kwargs"] = {"skip_download": True}
    ...: 
    ...:    # Many keyword arguements are required, as they provide information on
    ...:    # the names of coordinates needed to extract data
    ...:    input_kwargs["inst_lon_name"] = "glon"
    ...:    input_kwargs["mod_lon_name"] = "lon"
    ...:    input_kwargs["inst_name"] = ["glon", "glat", "altitude"]
    ...:    input_kwargs["mod_name"] = ["lon", "lat", "ZG"]
    ...:    input_kwargs["mod_units"] = ["deg", "deg", "cm"]
    ...:    input_kwargs["mod_datetime_name"] = "time"
    ...:    input_kwargs["mod_time_name"] = "time"
    ...: 
    ...:    # Now we are ready to run using the defaults!
    ...:    matched_inst = ps_mod.utils.match.collect_inst_model_pairs(stime, etime, tinc, cindi, **input_kwargs)
    ...: 
Mission acknowledgements and data restrictions will be printed here when available.
/Users/russellstoneback/opt/anaconda3/envs/standard2/bin/ipython:14: DeprecationWarning: Strict times will eventually be enforced upon all instruments. (strict_time_flag)
/Users/russellstoneback/opt/anaconda3/envs/standard2/bin/ipython:19: DeprecationWarning: Strict times will eventually be enforced upon all instruments. (strict_time_flag)

In [39]: matched_inst

In [40]:

Reply:
I also tried with,

filename = '/Volumes/BlackBox/ScienceData/ucar/tiegcm/tiegcm_icon_merg2.0_totTgcm.s_%j_%Y.nc'

Reply:
I can put up the TIEGCM file onto the UTD box service if you'd like to have a copy to work with. I've poked at the things I can think to poke at from outside.

Reply:
Does your pressure to altitude routine work as a pysat custom function?  Because that would take care of the altitude issue.

As for files, the non-kludge fix is for me to get one of the benchmark files from NCAR.  I tried to follow their instructions for globus, which didn't work, so have contacted their help.